### DISCUSSION TITLE
Pixel 6 won't connect to gnb

### DISCUSSION THREAD

**Discussion by ffxxcad** [srsran/srsRAN_Project#146] [General Help] [answered]
Hello everyone,

I've seen on this discussion board that people have had success [connecting Google Pixel 6 phones using N78 at 40 Mhz](https://github.com/srsran/srsRAN_Project/discussions/36#discussioncomment-5456973). However we have not had any success in the pixel 6 connecting to the gnb despite using similar configuration settings.
### The issue

The phone can see the network (shows up as 00101), but cannot join it:
![connection](https://github.com/srsran/srsRAN_Project/assets/132515083/fc00331a-5bf2-446a-8840-39cb83101550)

Network Signal Guru sees the network as well:
![NSG](https://github.com/srsran/srsRAN_Project/assets/132515083/3513a7a5-7d38-4a31-ab40-824c10e61a94)

I've double checked that the subscriber has been added in Open5GS and that the APN settings match that of the core:
![apn_settings](https://github.com/srsran/srsRAN_Project/assets/132515083/fc31b755-6e91-425a-a288-2c1c2b6f9d36)

Any thoughts of what might be going wrong?
### Setup

- Ettus USRP B210
- National Instruments CDA 2990
- Google Pixel 6 (rooted)
- Gialer Programable Sim Card
- Open5GS Core
### Files
[gnb_GP6.zip](https://github.com/srsran/srsRAN_Project/files/12176797/gnb_GP6.zip)
[gNB_mac.zip](https://github.com/srsran/srsRAN_Project/files/12176800/gNB_mac.zip)
[gNB_ngap.zip](https://github.com/srsran/srsRAN_Project/files/12176801/gNB_ngap.zip)
[amf_log.txt](https://github.com/srsran/srsRAN_Project/files/12176803/amf_log.txt)
[gNB_log.txt](https://github.com/srsran/srsRAN_Project/files/12176804/gNB_log.txt)

**Comment by robertfalkenberg** [comment]
It could be a performance issue. In the gNB.log, I see some warnings like
```
2023-07-26T19:17:39.060131 [GNB     ] [W] CPU19 scaling governor is not set to performance, which may hinder performance. You can set it to performance using the "srsran_performance" script
2023-07-26T19:17:39.060137 [GNB     ] [W] DRM KMS polling is enabled, which may hinder performance. You can disable it using the "srsran_performance" script
...
2023-07-26T19:18:27.706197 [FAPI    ] [W] [     0.0] Real-time failure in FAPI: Received late DL_TTI.request. Current slot is 705.11 while message corresponds to 705.10
2023-07-26T19:18:27.706220 [Low-PHY ] [W] Real-time failure in low-phy: Downlink data late for sector 0 and slot 704.10.
2023-07-26T19:18:27.706240 [Low-PHY ] [W] Real-time failure in low-phy: Downlink data late for sector 0 and slot 704.12.
2023-07-26T19:18:27.706449 [RF      ] [W] Real-time failure in RF: late
2023-07-26T19:18:27.706459 [RF      ] [W] Real-time failure in RF: late
2023-07-26T19:18:27.706529 [RF      ] [W] Real-time failure in RF: late
2023-07-26T19:18:27.706537 [Low-PHY ] [W] Real-time failure in low-phy: PRACH request late for sector 0, slot 705.18 and start symbol 0.
2023-07-26T19:18:27.706543 [RF      ] [W] Real-time failure in RF: late
2023-07-26T19:18:27.706549 [RF      ] [W] Real-time failure in RF: late
```

you could also try smaller bandwidth for the start. The device supports downto 20MHz (but not 10 MHz) in n78.
```
...
  srate: 23.04
...
  channel_bandwidth_MHz: 20
...
```

**Comment by ismagom** [comment]
Did you run the srsran_performance script?

**Comment by ffxxcad** [comment]
@ismagom & @robertfalkenberg,

Thanks for your replies.

I ran the srsran_performance script and tried out the 20 MHz but am still seeing the same behavior:

[amf.zip](https://github.com/srsran/srsRAN_Project/files/12187728/amf.zip)
[upf.zip](https://github.com/srsran/srsRAN_Project/files/12187729/upf.zip)

[gNB_mac.zip](https://github.com/srsran/srsRAN_Project/files/12187731/gNB_mac.zip)
[gNB_ngap.zip](https://github.com/srsran/srsRAN_Project/files/12187732/gNB_ngap.zip)

[pixel_gnb_config.zip](https://github.com/srsran/srsRAN_Project/files/12187737/pixel_gnb_config.zip)

[amf_log.txt](https://github.com/srsran/srsRAN_Project/files/12187738/amf_log.txt)
[gNB_log.txt](https://github.com/srsran/srsRAN_Project/files/12187739/gNB_log.txt)

I've been able to successfully connect a Oneplus8T phone on N78 and N25 so I feel confident that srsRAN is working properly. Seems like there is something going on with the phone but I can't tell what it is.

**Reply by robertfalkenberg** [reply]
Does the IMSI of the uSIM you use in the Google Pixel 6 start with 00101... as well?

**Reply by robertfalkenberg** [reply]
Here is a list of PLMN/IMSI combinations that I have tested:
https://github.com/srsran/srsRAN_Project/discussions/100#discussioncomment-6496164

Please also ensure the system software is at least TQ3A.230605.010, before that it just refused to connect:
https://github.com/srsran/srsRAN_Project/discussions/100#discussioncomment-6452319

**Comment by robertfalkenberg** [comment]
Here is a list of PLMN/IMSI combinations that I have tested:
https://github.com/srsran/srsRAN_Project/discussions/100#discussioncomment-6496164

Please also ensure the system software is at least TQ3A.230605.010, before that it just refused to connect:
https://github.com/srsran/srsRAN_Project/discussions/100#discussioncomment-6452319

**Comment by ffxxcad** [comment, marked as answer]
Alright so I tried updating to TQ3A.230605.010 as suggested by @robertfalkenberg above and it still didn't resolve the issue.

I also tried changing the ims settings as suggested by @robertfalkenberg [here](https://github.com/srsran/srsRAN_Project/discussions/100#discussioncomment-6260974) and still no luck.

I triple checked the SIM settings and noticed that the SIM I was using hadn't been completely programmed to 00101. The plmn settings in the yellow boxes below were 33102.
![SIM_Setup](https://github.com/srsran/srsRAN_Project/assets/132515083/95aa9eca-44ca-432d-8471-be2e6e62e107)
These values never affected the Oneplus8T phone's ability to connect so I had never updated them. After changing the plmn values to 00101 the phone was able to connect to the core and worked great on both SISO and 2x2 MIMO.

tl;dr: make sure that **every** instance of plmn is 00101 and it should work.

For anyone who may be interested here are the files from the successful test:
[gNB_log.zip](https://github.com/srsran/srsRAN_Project/files/12241309/gNB_log.zip)
[gNB_mac.zip](https://github.com/srsran/srsRAN_Project/files/12241315/gNB_mac.zip)
[gNB_ngap.zip](https://github.com/srsran/srsRAN_Project/files/12241316/gNB_ngap.zip)

And the used config files:
[amf.zip](https://github.com/srsran/srsRAN_Project/files/12241323/amf.zip)
[gnb_GP6_Config.zip](https://github.com/srsran/srsRAN_Project/files/12241324/gnb_GP6_Config.zip)
[upf.zip](https://github.com/srsran/srsRAN_Project/files/12241325/upf.zip)

Thanks again for the suggestions @robertfalkenberg, they got me on the right path.

**Reply by robertfalkenberg** [reply]
Hi @ffxxcad ,
thanks for taking the time to share your experience with the community, especially for the important note with the `EF_*PLMN*` fields.
Since we mostly customize our SIM cards with pySim-prog (osmocom), the mentioned fields are automatically maintained when the IMSI is changed. Of course, this does not have to be the case with other tools.

**Reply by mohit4444** [reply]
Hi @robertfalkenberg . Could you please help me with this one? https://github.com/srsran/srsRAN_Project/discussions/1205#discussion-8317426
